src/AmsterdamPHP/Bundle/SiteBundle/Resources/config/services.yml

```diff
@@ -1,10 +1,30 @@
+parameters:
+    amsterdamphp_site.integration.youtube_factory.class: AmsterdamPHP\Bundle\SiteBundle\Service\YoutubeFactory
+    amsterdamphp_site.integration.youtube_client.class: Madcoda\Youtube
+
 services:
+    amsterdamphp_site.integration.youtube_factory:
+        class: %amsterdamphp_site.integration.youtube_factory.class%
+        arguments:
+            key: %youtube_key%
+
+    amsterdamphp_site.integration.youtube_client:
+        class:            "%amsterdamphp_site.integration.youtube_client.class%"
+        factory_service:  amsterdamphp_site.integration.youtube_factory
+        factory_method:   getClient
```

Comment:
After talking with you this was my assumption as how you preferred to have it setup. (Heck I took you meetup bundle as example.) If this is fine I'll let it be but if you really prefer instantiating the client directly I'll change it to that.